nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -468,6 +475,28 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in, const std_msgs::msg:
     // Update polygon coordinates
     polygon->updatePolygon(cmd_vel_in);
 
+    // Get collision points for sources associated to polygon
+    std::vector<std::string> sources_names = polygon->getSourcesNames();
```

Comment:
It seems to me to be more sensible to pass in a reference to the map of collision points to `isPointsInside` function for each of the polygons and let each of the polygons use its own internal `getSourcesNames` when iterating through the map to decide if it wants to use it or not.

In that case, this block of code would be removed and we just pass in the map of collision points rather than the vector of collision points to the `processXYZ()` functions